src/main/scala/sangria/macros/derive/DeriveObjectTypeMacro.scala

```diff
@@ -105,9 +106,14 @@ class DeriveObjectTypeMacro(context: blackbox.Context) extends {
             case (acc, _) ⇒ acc
           }
 
+          val objectTypeName: c.universe.Tree = config.collect{case MacroTransformFieldNames(fnt) => fnt}.lastOption match {
+            case Some(fnt) => q"$fnt(${configName orElse annotationName getOrElse q"$name"})"
+            case None => configName orElse annotationName getOrElse q"$name"
```

Comment:
maybe we can extract `configName orElse annotationName getOrElse q"$name"` in a val, just to avoid the code duplication?